bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,921 @@
+    const timeEntries = {
+        id: timeEntryId ? parseInt(timeEntryId) : parseInt(timeSheetRecord),
+        ts_id: timeSheetRecord,
+        ts_date: new Date(parseInt(timeSheetDateTimeStamp)),
+        company_code: companyCode,
+        person_work_agreement: personWorkAgreement,
+        person_work_agreement_external_id: personWorkAgreementExternalID,
+        ts_df_controlling_area: controllingArea,
+        ts_df_hours_unit_of_measure: hoursUnitOfMeasure,
+        ts_df_purchase_order_item: purchaseOrderItem,
+        ts_df_receiver_cost_center: receiverCostCenter,
+        ts_df_recorded_hours: recordedHours,
+        ts_df_recorded_quantity: recordedQuantity,
+        ts_df_ts_overtime_category: timeSheetOvertimeCategory,
+        ts_df_ts_task_component: timeSheetTaskComponent,
+        ts_df_ts_task_level: timeSheetTaskLevel,
+        ts_df_ts_task_type: timeSheetTaskType,
+        ts_df_ts_note: timeSheetNote,
+        ts_operation: timeSheetOperation,
+        ts_status: timeSheetStatus,
+        user_email: user_email
+    }
+
+    if (timeClockManagement && startedAt !== "undefined") {
+        timeEntries.start_at = startedAt ?? moment().format()
+        timeEntries.end_at = timeEntryId ? endAt.format() : ''
+    }
+
+    dataStore.save("TimeEntries", timeEntries);
+    dataStore.save("TimeEntriesClock", timeEntries);
+}
+
+function timeToDecimal(timeStr) {
+    const [hours, minutes] = timeStr.split(':')
+    return parseInt(hours, 10) + parseInt(minutes, 10) / 60;
+}
+
+function fullSync({dataStore, client}) {
+    return getTimeEntries(dataStore, client);
+}
+
+async function getEmployeeBusinessPartner(client, user_email) {
+    const employeeDataUrl = `YY1_EMPLOYEEDATA_CDS/YY1_EmployeeData?$filter=DefaultEmailAddress eq '${user_email}'`
+
+    const employeeDataGET = await client.fetch(employeeDataUrl, defaultGetOptions);
+
+    if (!employeeDataGET.ok) {
+        throw new Error(`Could not Retrieve Employee Data(${employeeDataGET.status}: ${employeeDataGET.statusText})`);
+    }
+
+    const employeeDataResponse = await employeeDataGET.json()
+
+    return employeeDataResponse.d.results[0].BusinessPartner
+}
+
+async function getWorkAgreement(client, employeeBusinessPartner) {
+    const workAgreementUrl = `YY1_WORKAGREEMENTDETAILS_CDS/YY1_WorkAgreementDetails?$filter=Person eq '${employeeBusinessPartner}'`
+
+    const workAgreementGET = await client.fetch(workAgreementUrl, defaultGetOptions);
+
+    if (!workAgreementGET.ok) {
+        throw new Error(`Could not Retrieve Employee Data(${workAgreementGET.status}: ${workAgreementGET.statusText})`);
+    }
+
+    const workAgreementResponse = await workAgreementGET.json()
+
+    return workAgreementResponse.d.results[0]
+}
+
+function incrementalSync({dataStore, client, latestSynchronizationTime}) {
+    return getTimeEntries(dataStore, client, latestSynchronizationTime);
+}
```

Comment:
`latestSynchronizationTime` is not used in `getTimeEntries()`
It should be called this way:

```suggestion
function fullSync(args) {
    return getTimeEntries(args, false);
}

function incrementalSync(args) {
    return getTimeEntries(args, true);
}

async function getTimeEntries({dataStore, client}, incrementalSync) {
    // ...
    const startTime = incrementalSync ? sevenDaysBefore : thirtyDaysBefore;    
    const dateFilter = `TimeSheetDate ge datetime'${startTime}' and TimeSheetDate le datetime'${today}'`;
    
    // or maybe better without endtime?
    const dateFilter = `TimeSheetDate ge datetime'${incrementalSync ? sevenDaysBefore : thirtyDaysBefore}'`;
```

Reply:
code suggestion applied